Forms: deprecate outlined and animated form styles.

ramonjd opened the pull request:
Related to https://github.com/Automattic/jetpack/pull/43327

## Proposed changes:

Deprecates Outlined and Animated styles (in fact, all styles) and removes them when a legacy form is loaded in the editor.

 When a form is opened using this branch, the `is-style-outline` is detectted  upon load. 

Only if you save the editor does the change take place on the frontend.

<img width="762" alt="Screenshot 2025-05-06 at 2 46 19 pm" src="https://github.com/user-attachments/assets/5d13c6e5-96e6-4bc2-9f7e-95665f90e9f9" />



### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

ramonjd committed:
Deprecate automatically

ramonjd commented on projects/packages/forms/src/blocks/contact-form/edit.js:
```diff
@@ -103,6 +104,8 @@ function JetpackContactFormEdit( { name, attributes, setAttributes, clientId, cl
 	const blockProps = useBlockProps( { ref: wrapperRef } );
 	const formClassnames = clsx( className, 'jetpack-contact-form' );
+	const hasUpdatedFormStyle = useHasDeprecatedFormStyle( clientId );
+	console.log( 'hasUpdatedFormStyle', hasUpdatedFormStyle );
```
Remove this. 😄

ramonjd commented on projects/packages/forms/src/blocks/contact-form/index.js:
```diff
@@ -96,13 +96,16 @@ export const settings = {
+	// { name: 'animated', label: __( 'Animated', 'jetpack-forms' ) },
+	// @deprecated
+	// { name: 'outlined', label: __( 'Outlined', 'jetpack-forms' ) },
+	// @TODO check if this is needed. It was added 2 years ago and never activated. Time to delete associated styles?
```
Let's get rid of the "below" style and associated code in another PR.

ramonjd commented on projects/packages/forms/src/blocks/contact-form/util/form.js:
```diff
@@ -61,3 +62,35 @@ export const useFormStyle = clientId => {
 	return getBlockStyle( formBlockAttributes?.className ) || FORM_STYLE.DEFAULT;
 };
+
+export const useHasDeprecatedFormStyle = formBlockClientId => {
```
Optionally, we could add an embedded alert in each form with a legacy style (not this one), but just to give you an idea:

<img width="771" alt="Screenshot 2025-05-05 at 4 43 47 pm" src="https://github.com/user-attachments/assets/e4ffae5e-c409-426d-9464-dbba5415cfe4" />

ramonjd commented on projects/packages/forms/src/blocks/contact-form/index.js:
```diff
@@ -96,13 +96,16 @@ export const settings = {
+	// @deprecated
+	// { name: 'animated', label: __( 'Animated', 'jetpack-forms' ) },
+	// @deprecated
+	// { name: 'outlined', label: __( 'Outlined', 'jetpack-forms' ) },
```
To keep things need we could extricate all styles and code related to the styles and place them in a deprecated folder.

talldan commented:
> What you're seeing in this screencast is a form built in trunk using the Outlined style. When opened using this branch, the is-style-outline is remove automatically upon load.

I personally think we can retain the class, so it becomes only a custom class name that the user can still update in the 'advanced' block tools (I think the classnames are already visible there, so I think it already works). 

Otherwise the user might be making a small unrelated edit to their form (e.g. changing another block or the title of their post), and then find their form styles are changed when saving.

I think the only complex thing would be if we introduce other style variations in the future, and then the user ends up mixing `is-style-outlined` with another block style.

ramonjd commented:
> I personally think we can retain the class, so it becomes only a custom class name that the user can still update in the 'advanced' block tools (I think the classnames are already visible there, so I think it already works).

This is what I originally did. The form looked completely borked.

I think it's because when removing the styles from the registration data in index.js, the block state no longer knows if the form has the style or not.  And because the style determines the HTML, e.g., it conditionally loads all the notched HTML, everything explodes.

I haven't spent much time yet looking around, but I suppose we could detect the styles on the className attributes and load the notched HTML.

I'll try that tomorrow.

talldan commented:
> This is what I originally did. The form looked completely borked.

In the editor or frontend? I think in the editor we could consider completely removing the `is-style-outlined` / `is-style-animated` css along with that code that adds the extra markup. The editor wouldn't look like the front-end, but that's the benefit to us of deprecating it IMO. Or alternatively we keep some similar styles that look close, but not completely the same.

On the frontend I think it's this code here - https://github.com/Automattic/jetpack/blob/be8aad40ccaa1406bd78a88d92f7daf83280e085/projects/packages/forms/src/contact-form/class-contact-form-field.php#L1462-L1478

ramonjd commented:
> In the editor or frontend? 

The editor. So removing the notched HTML and associated style CSS in the editor should work. 

We'll leave the frontend as is of course.

ramonjd commented:
> I think in the editor we could consider completely removing the is-style-outlined / is-style-animated css along with that code that adds the extra markup. The editor wouldn't look like the front-end, but that's the benefit to us of deprecating it IMO.

I tried retaining the style classname (`is-style-outline` and `is-style-animated`) on the form block where previously applied, but it's a bit confusing as there's no obvious way for the user to remove it to make the editor and frontend consistent. 

The editor and frontend will continue to look differently until the user removes the classname from the advance panel in the form's block settings.

Maybe it's just me, and it's not a problem. 😄 

Anyway, here I've removed the styles from the registration object, and removed all **editor** CSS and HTML related to the animated/outline styles. 

https://github.com/user-attachments/assets/9261b59c-4167-4606-b08e-540d97f26d3d

I generally think it's okay, but it might require some nudge in the editor? What do folks think?

ramonjd commented:
> The editor and frontend will continue to look differently until the user removes the classname from the advance panel in the form's block settings.
> 
> Maybe it's just me, and it's not a problem. 😄

Withdraw that. It is a problem.

I think what we're missing with this solution is the issue we were trying to fix in the first place with https://github.com/Automattic/jetpack/pull/43366: if you change the border styles of input blocks, they won't be reflected on the frontend. 

So I think we're compounding confusion here: we're deprecating the style, keeping the class name around in the editor, but retaining all the bugs.

talldan commented:
> Withdraw that. It is a problem.

> I think what we're missing with this solution is the issue we were trying to fix in the first place with https://github.com/Automattic/jetpack/pull/43366: if you change the border styles of input blocks, they won't be reflected on the frontend.

> So I think we're compounding confusion here: we're deprecating the style, keeping the class name around in the editor, but retaining all the bugs.

I guess we could keep the frontend solution from the other PR to keep borders working, since that wasn't as complex as the editor solution. 🤔 

Another addition could be a small notice in the style variations UI explaining that a deprecated style is being used (hopefully there's a inspector controls group for that). The notice could also provide the option to remove the classname through a button.

Maybe you're right though, the experience could be too broken to try deprecating these styles.

I do think exploring this PR gives a better idea of the trade offs. The situation with these styles is pretty messy, thanks for continuing to look into it!

ramonjd commented:
> Another addition could be a small notice in the style variations UI explaining that a deprecated style is being used (hopefully there's a inspector controls group for that). The notice could also provide the option to remove the classname through a button.

That's a good compromise, at least for this PR. I'll update accordingly so folks can compare. 

Mentioned on the other PR, but a refactor is possible with a few trade offs (label needs a background color to "break" the input border):

https://codepen.io/ramonjd/pen/ByyxExd

ramonjd committed:
Implement deprecation notice for outdated form styles in contact form block

ramonjd commented:
parking this. 

we can keep for future reference/discussion prompting over tea and cake
